app/controllers/api/v1/search_results_controller.rb

```diff
@@ -3,13 +3,8 @@
 module Api
   module V1
     class SearchResultsController < ApplicationController
-      before_action :doorkeeper_authorize!
-
       def create
-        upload_form = UploadForm.new(
-          search_engine: upload_form_params[:search_engine],
-          csv_file: upload_form_params[:csv_file]
-        )
+        upload_form = create_upload_form
```

Comment:
This is an anti-pattern which I have raised before. Prefer the following:

```
def creates
  if upload_form.valid?
		##...
end 

private

def upload_form
 @upload_form ||= UploadForm.new(
 #...
 )
end
```

This operator will return the existing instance if it is already defined. 

In Ruby, the principle "[tell, don't ask](https://thoughtbot.com/blog/tell-dont-ask)" should be followed.